src/pkg/storage/storage.go

```diff
@@ -166,13 +171,71 @@ func mustMatchConfig(
 		if !ok {
 			continue // m may declare values which aren't stored in the config file
 		}
-
 		vv := cast.ToString(g.Get(tomlK))
-		if v != vv {
+
+		areEqual := false
+
+		if IsValidPath(v) && IsValidPath(vv) {
+			areEqual, _ = ArePathsEquivalent(v, vv)
+		} else {
+			areEqual = v == vv
+		}
+
+		if !areEqual {
 			err := clues.New("value of " + k + " (" + v + ") does not match corso configuration value (" + vv + ")")
 			return clues.Stack(ErrVerifyingConfigStorage, err)
 		}
 	}
 
 	return nil
 }
+
+func ArePathsEquivalent(path1, path2 string) (bool, error) {
+	normalizedPath1 := strings.TrimSpace(filepath.Clean(path1))
+	normalizedPath2 := strings.TrimSpace(filepath.Clean(path2))
+
+	normalizedPath1 = strings.TrimSuffix(normalizedPath1, string(filepath.Separator))
+	normalizedPath2 = strings.TrimSuffix(normalizedPath2, string(filepath.Separator))
+
+	return normalizedPath1 == normalizedPath2, nil
+}
+
+func IsValidPath(path string) bool {
+	if _, err := os.Stat(path); err == nil {
+		return true
+	}
+
+	isDir := strings.TrimSpace(filepath.Ext(path)) == ""
+
+	if isDir {
+		err := os.Mkdir(path, 0644)
+		if err == nil {
+			os.Remove(path)
```

Comment:
why are we removing the paths we just created?

Reply:
@ryanfkeepers The directory and files are being created as an empirical way of testing if they are valid or not. So it is not required to create those paths but rather to check if they can be created and therefore qualifies to be a valid path.

Reply:
Hmm, I'm a little dubious about that handling, since it has the potential to leave behind dirty state under certain conditions.  Is there a way to get the same assurances without creation?

Actually, are we already auto-generating config file locations if they don't exist?  If we aren't, can we reduce the assertion to only ensure the specified location is already present?